### PR TITLE
[4.0] Implementation of disable in joomla-field-switcher

### PR DESCRIPTION
**Pull request by astridx**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/20852.

### Summary of Changes

- Add  attributes to `joomla-field-switcher` if field is set to `disable="true"`

- Make the field `joomla-field-switcher` if field is set to `disable="true"`

### Testing Instructions

Open global configuration and see outer html of the field "Search Engine Friendly URLs"
```

<joomla-field-switcher id="jform_sef" off-text="No" on-text="Yes" type="success">
<span class="switcher active has-success" tabindex="0" aria-checked="true" role="switch" aria-label="Yes"><input id="jform_sef0" name="jform[sef]" value="0" tabindex="-1" class="valid form-control-success" aria-invalid="false" type="radio"><input id="jform_sef1" name="jform[sef]" value="1" class="valid active form-control-success" tabindex="-1" checked="" aria-invalid="false" type="radio"><span class="switch"></span></span><span class="switcher-labels"><span class="switcher-label-0">No</span><span class="switcher-label-1 active">Yes</span></span></joomla-field-switcher>
```

Open `/administrator/components/com_config/forms/application.xml`

and change the field (insert `disabled="true"`)

```
		<field
			name="sef"
			type="radio"
			label="COM_CONFIG_FIELD_SEF_URL_LABEL"
			class="switcher"
			default="1"
			filter="boolean"
			>
			<option value="0">JNO</option>
			<option value="1">JYES</option>
		</field>

```
to 

```
		<field
			name="sef"
			type="radio"
			label="COM_CONFIG_FIELD_SEF_URL_LABEL"
			class="switcher"
			disabled="true"
			default="1"
			filter="boolean"
			>
			<option value="0">JNO</option>
			<option value="1">JYES</option>
		</field>
```

Open again global configuration and see outer html of the field "Search Engine Friendly URLs". Now the text `disabled=""`  is added.

Apply this patch and see, that the text `disabled="disabled"`  is added to the `joomla-field-switcher` and the `inputs` of  `joomla-field-switcher`. See also, that the field now works as expected.

See also Issue https://github.com/joomla/joomla-cms/issues/20852.

### Expected result
The field "Search Engine Friendly URLs" is disabled. You can not change it.

### Actual result
Nothing changed after making the field disabled. The field works as before. You can change the value. 

### Documentation Changes Required
Yes.

### CODE REVIEW

File: build/webcomponents/scss/field-switcher/field-switcher.scss [hunk 1]
```diff
@@ -105,6 +105,11 @@ joomla-field-switcher {
   }
 }
 
+// Make transparent if disabled
+joomla-field-switcher[disabled="disabled"] .switcher {
```
[the remark this answers is not in the document]
Comment (ciar4n) [reply build/webcomponents/scss/field-switcher/field-switcher.scss:109]: I think we need to change the hound settings here. It is unavoidable qualifying attribute selectors with a custom element unless we start adding classes to the element..

```
<joomla-field-switcher class="joomla-field-switcher">
 ..
</joomla-field-switcher>
```

**Comment by C-Lodder** [comment]
Are you this prevents being able to switch the switcher using the `Enter` and `Esc` keys on the keyboard?

If not, check the solution I provided in the issue

**Comment by astridx** [comment]
@C-Lodder Thanks. Your are right, I forgot to disabling the `keys`. I will have a look later.

A question ofttopic: What files should I put into this pr? only the js and css file build/webcomponents/ or the files after running `node build.js --compilecescss` and `node build.js --compilecejs` in the media folder?

File: build/webcomponents/scss/field-switcher/field-switcher.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (astridx) [reply build/webcomponents/scss/field-switcher/field-switcher.scss:109]: Should I create the class joomla-field-switcher?

**Comment by astridx** [comment]
Now this prevents being able to switch the switcher using the Enter and Esc keys on the keyboard. 

What should I do with the hound violation?

**Comment by dgrammatiko** [comment]
@astridx can you please do the PR here: https://github.com/joomla/40-backend-template/pull/395

We're moving to web components for the simple fields

**Comment by ciar4n** [comment]
I would say ignore the violation.

**Comment by astridx** [comment]
I created the PR here https://github.com/joomla/40-backend-template/pull/445

Should I close this PR?

**Comment by laoneo** [comment]
Closing as the pr got moved to the back end template repo.

**Pull request closed without merging**